billing/collectmessage.go

```diff
@@ -0,0 +1,186 @@
+
+}
+
+// TODO : When the source object is inconsistent with the target object, update the entire object information
+func delCopyObject(msg messagebus.ConsumerMessage) {
+	Logger.Println("[MESSAGE] Enter del Copy object, Uuid is:", msg.Uuid)
+	info := msg.Messages
+	object := new(db.BillingInfo)
+	object.ProjectId = info[messagebus.ProjectId]
+	object.BucketName = info[messagebus.BucketName]
+	object.ObjectName = info[messagebus.ObjectName]
+	object.CreateTime, _ = time.Parse(db.TIME_LAYOUT_TIDB, info[messagebus.TimeLocal])
+	v := math.MaxUint64 - uint64(object.CreateTime.UnixNano())
+	object.Version = strconv.FormatUint(v, 10)
+	objectSize, _ := strconv.ParseInt(info[messagebus.ObjectSize], 10, 64)
+	object.Size = objectSize
+	object.CountSize = getCountSize(object.Size)
+	srcStorageClass := StorageClassStringMap[info[messagebus.StorageClass]]
+	targetStorageClass := StorageClassStringMap[info[messagebus.TargetStorageClass]]
+	if srcStorageClass == 0 {
+		object.StorageClass = uint8(targetStorageClass)
+		err := db.DbClient.InsertBilling(*object)
+		if err != nil {
+			Logger.Println("[MESSAGE ERROR] Del Copy Object Error with database insert, err:", err, "Uuid is :", msg.Uuid)
+			return
+		}
+	} else if targetStorageClass == 0 {
+		err := db.DbClient.DeleteBillingObject(*object)
+		if err != nil {
+			Logger.Println("[MESSAGE ERROR] Del Copy Object Error with database delete, err:", err, "Uuid is :", msg.Uuid)
+			return
+		}
+	} else {
+		object.StorageClass = uint8(targetStorageClass)
+		err := db.DbClient.UpdateBilling(*object)
+		if err != nil {
+			Logger.Println("[MESSAGE ERROR] Del Copy Object Error with database update, err:", err, "Uuid is :", msg.Uuid)
+			return
+		}
+	}
+	Logger.Println("[MESSAGE] Copy object successful, Uuid is:", msg.Uuid)
+}
+
+// TODO
+func delCopyObjectPart(msg messagebus.ConsumerMessage) {
+
+}
+
+// TODO
+func delRenameObject(msg messagebus.ConsumerMessage) {
+
+}
+
+// TODO
+func delPostObject(msg messagebus.ConsumerMessage) {
+
+}
+
+func delDeleteObject(msg messagebus.ConsumerMessage) {
+	Logger.Println("[MESSAGE] Enter del delete object, Uuid is:", msg.Uuid)
+	info := msg.Messages
+	object := new(db.BillingInfo)
+	object.ProjectId = info[messagebus.ProjectId]
+	object.BucketName = info[messagebus.BucketName]
+	object.ObjectName = info[messagebus.ObjectName]
+	object.CreateTime, _ = time.Parse(db.TIME_LAYOUT_TIDB, info[messagebus.TimeLocal])
+	object.Deleted = Deleted
+	object.DeleteTime = object.CreateTime.AddDate(0, 1, 0)
+	err := db.DbClient.UpdateBillingFlag(*object)
+	if err != nil {
+		Logger.Println("[MESSAGE ERROR] Del Delete Object Error with database update, err:", err, "Uuid is :", msg.Uuid)
+		return
+	}
+	Logger.Println("[MESSAGE] Delete object successful, Uuid is:", msg.Uuid)
+}
+
+// Round up for the smallest unit
+func getCountSize(objectSize int64) (countSize int64) {
+	countSize = objectSize - (objectSize % MinSizeForStandardIa) + MinSizeForStandardIa
```

Comment:
You mean `countSize = (objectSize + MinSizeForStandardIa - 1) / MinSizeForStandardIa * MinSizeForStandardIa`?